e2e/e2e.py

```diff
@@ -0,0 +1,306 @@
+        time.sleep(USER_RESPONSE_TIME)
+        wait = WebDriverWait(driver, 10) # wait up to 10s to find the element
+        button = wait.until(EC.presence_of_element_located((By.CSS_SELECTOR, selector)))
+        button.click()
+    except Exception as e:
+        print("Failed when clicking on button '{}'".format(selector))
+        raise e
+
+def click_button_after_hover(driver, hover_selector, button_selector):
+    try:
+        wait = WebDriverWait(driver, 10) # wait up to 10s to find the element
+        hover = wait.until(EC.presence_of_element_located((By.CSS_SELECTOR, hover_selector)))
+        actions = ActionChains(driver)
+        actions.move_to_element(hover).perform()
+        driver.find_element(By.CSS_SELECTOR, button_selector).click()
+    except Exception as e:
+        print("Failed when hovering over '{}' and clicking on button '{}'".format(hover_selector, button_selector))
+        raise e
+
+def select(driver, selector, selection):
+    try:
+        wait = WebDriverWait(driver, 10) # wait up to 10s to find the element
+        element = Select(wait.until(EC.presence_of_element_located((By.CSS_SELECTOR, selector))))
+        element.select_by_visible_text(selection)
+    except Exception as e:
+        print("Failed when selecting from drop down '{}'".format(selector))
+        raise e
+
+def check_presence_of(driver, selector):
+    try:
+        wait = WebDriverWait(driver, 10)
+        element = wait.until(EC.presence_of_element_located((By.CSS_SELECTOR, selector)))
+    except Exception as e:
+        print("failed when checking presence of element '{}'".format(selector))
+        raise e
+
+def register(driver, email, username, password):
+    """Registers the user by filling in the registration form"""
+    # Access registration page
+    click_button(driver, "#register")
+
+    # Fill in registration form
+    fill_field(driver, "#email", email)
+    fill_field(driver, "#username", username)
+    fill_field(driver, "#password", password)
+
+    # Click on registration button
+    click_button(driver, "#registerButton")
+
+    # Click on finalize verification button
+    click_button(driver, "#finalizeVerification")
+
+def access_game_list(driver):
+    """Accesses the list of games"""
+    # Go back to the main page
+    click_button(driver, "#home")
+
+    # Go to the lobby
+    click_button(driver, "#seeGameList")
+
+    # Now I should see the game list (this will throw it if does not find it)
+    check_presence_of(driver, '#actualGames')
+
+def logout(driver):
+    """Logs the current user out"""
+    # Just click on the logout button
+    click_button_after_hover(driver, "#connectedUserName", "#logout")
+
+def login(driver, email, password):
+    # Go to the login page
+    click_button(driver, "#login")
+
+    # Fill in the form
+    fill_field(driver, "#email", email)
+    fill_field(driver, "#password", password)
+
+    # Log in
+    click_button(driver, "#loginButton")
+
+def scenario(kind):
```

Comment:
plus explicite le nom ? c'pas clair

Reply:
c'est une annotation, pour dire `@scenario(kind)` avant un scénario de test. T'as un autre nom à proposer ?

Reply:
Je comprends pas ta réponse non plus :p !

Reply:
cette fonction permet d'annoter d'autres fonctions (lignes @scenario) qui elles définissent chacune un scénario de test. Cela permet de facilement les collecter pour itérer dessus